test/cloudtest/test_storaged_shared_fate.py

```diff
@@ -0,0 +1,219 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import subprocess
+import time
+from textwrap import dedent
+
+from materialize.cloudtest.application import MaterializeApplication
+
+CLUSTER_SIZE = 8
+NUM_SOURCES = 4
+
+
+def populate(mz: MaterializeApplication, seed: int) -> None:
+    create_sources = "\n".join(
+        f"""
+            > CREATE SOURCE s{i}
+              IN CLUSTER storaged_shared_fate
+              FROM KAFKA CONNECTION kafka (TOPIC 'testdrive-storaged-shared-fate-${{testdrive.seed}}')
+              FORMAT BYTES
+              ENVELOPE NONE;
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    create_views = "\n".join(
+        f"""
+            > CREATE MATERIALIZED VIEW v{i} AS SELECT COUNT(*) FROM t1 UNION ALL SELECT COUNT(*) FROM s{i};
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    create_indexes_and_counts = "\n".join(
+        f"""
+            > CREATE DEFAULT INDEX ON v{i};
+
+            > SELECT COUNT(*) > 0 FROM s{i};
+            true
+    """
+        for i in range(NUM_SOURCES)
+    )
+
+    mz.testdrive.run(
+        # TODO: I need both clusters? Can't create index/mv in cluster which has sources connected with IN CLUSTER
+        input=dedent(
+            f"""
+            > CREATE CLUSTER storaged_shared_fate REPLICAS (storaged_shared_fate_replica (SIZE '{CLUSTER_SIZE}-1'));
+
+            > CREATE CLUSTER shared_fate REPLICAS (shared_fate_replica (SIZE '{CLUSTER_SIZE}-1'));
+
+            > SET cluster = storaged_shared_fate;
+
+            > CREATE CONNECTION kafka TO KAFKA (BROKER '${{testdrive.kafka-addr}}')
+
+            $ kafka-create-topic topic=storaged-shared-fate partitions={CLUSTER_SIZE}
+
+            $ kafka-ingest format=bytes topic=storaged-shared-fate repeat=1000
+            CDE${{kafka-ingest.iteration}}
+
+            {create_sources}
+
+            > CREATE TABLE t1 (f1 INTEGER);
```

Comment:
I think there is no need to use tables in this test. Tables are an `environmentd`-side concept ,that is, environmentd will directly put stuff in S3 without involving a storage cluster. Having a source with kafka messages is what is exercising the storage cluster.